src/handleAction.js

```diff
@@ -7,19 +7,14 @@ export default function handleAction(type, reducers, defaultState) {
     ? type.toString()
     : type;
 
+  const nextReducer = isFunction(reducers) ? reducers : reducers.next;
+  const throwReducer = isFunction(reducers) ? reducers : reducers.throw;
+
   return (state = defaultState, action) => {
     // If action type does not match, return previous state
     if (action.type !== typeValue) return state;
 
-    const handlerKey = action.error === true ? 'throw' : 'next';
-
-    // If function is passed instead of map, use as reducer
-    if (isFunction(reducers)) {
-      reducers.next = reducers.throw = reducers;
-    }
-
-    // Otherwise, assume an action map was passed
-    const reducer = reducers[handlerKey];
```

Comment:
Couple ideas. The main one was, yeah, to move logic as high up in the scope as possible. I also wanted to get rid of the magic strings `'throw'` and `'next'`, the property accessor `reducers[handlerKey]` in favor of the more explicit `nextReducer`/`throwReducer`, and generally short circuit the whole process as much as possible. The fact that the new version has fewer lines of code and is (imo) simpler is an unintended but welcome side effect.